### PR TITLE
Validator exit bus oracle: tests & fixes

### PR DESCRIPTION
**Pull request by DiRaiks**
Smart contract changes:

- `ValidatorsExitBusOracle` now enforces that validator exit requests for the same (staking module, node operator) pair have increasing validator indices. This should be revisited after/if Ethereum implements validator index recycling, but that would require changing the exit bus logic anyway. 

Tests:

- [ ]  clean up/add helpers/etc at existing tests
- [x]  access control
    - [x]  Admin is set at initialise
    - [x]  `SUBMIT_DATA_ROLE`
    - [x]  `PAUSE_ROLE`
    - [x]  `RESUME_ROLE`
- [ ]  deploy tests
    - [ ]  initialise
        - [ ]  paused at initilazed
    - [ ]  interacting contracts are set
- [ ]  pause/resume(can be part of other tests)
    - [ ]  report submitting is not available at pause
- [ ]  `_handleConsensusReport` callback(invoked at the end of `baseOracle.submitConsensusReport` by `HashConsensus` contract)
    - [ ]  happy path
    - [ ]  emits warning on incomplete processing
- [ ]  `submitReportData`
    - [ ]  surface tests that baseOracle internal checks are invoked
    - [ ]  security(can be part of other/older tests)
        - [ ]  custom AC (`_checkMsgSenderIsAllowedToSubmitData`)
        - [ ]  hash match with report is enforced
        - [ ]  data safety is enforced
    - [ ]  `_handleConsensusReportData`
        - [ ]  enforces data format
        - [ ]  enforces length
        - [x]  enforces increasing validator indices within one node operator
        - [ ]  sanity check invoke
        - [ ]  `_processExitRequestsList`
            - [ ]  assembly
        - [ ]  processing state is updated
- [ ]  At Init/After/Between report view function reflect correct oracle state
    - [ ]  `getTotalRequestsProcessed`
    - [ ]  `getLastRequestedValidatorIndices`
    - [ ]  `getProcessingState`

### CODE REVIEW

**Comment by TheDZhon** [comment]
Gonna merge it tomorrow till noon UTC :pray: 
Let's coordinate the pivot point to spin up a new pull request.

File: test/0.8.9/oracle/validators-exit-bus-oracle-access-control.test.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        assert.isDefined(oracleVersion)
+        assert.isDefined(initTx)
+        assert.isDefined(exitRequests)
+        assert.isDefined(reportFields)
+        assert.isDefined(reportItems)
+        assert.isDefined(reportHash)
+      })
+    })
+    context('DEFAULT_ADMIN_ROLE', () => {
+      beforeEach(deploy)
```
Comment (TheDZhon) [inline test/0.8.9/oracle/validators-exit-bus-oracle-access-control.test.js:87]: I suggest using EVM snapshot here and there to speed-up the tests a little
```js
before() {
    // do all the stuff
    snapshot = new EvmSnapshot(hre.ethers.provider)
    await snapshot.make()
}

afterEach(async () => {
    await snapshot.rollback()
})
```

File: test/0.8.9/oracle/validators-exit-bus-oracle-access-control.test.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+      context('Admin is set at initialise', () => {
+        it('should set admin at initialise', async () => {
+          const DEFAULT_ADMIN_ROLE = await oracle.DEFAULT_ADMIN_ROLE()
+          await assert.emits(initTx, 'RoleGranted', { role: DEFAULT_ADMIN_ROLE, account: admin, sender: admin })
+        })
+
+        it('should revert without admin address', async () => {
+          await assert.reverts(
```
Comment (TheDZhon) [inline test/0.8.9/oracle/validators-exit-bus-oracle-access-control.test.js:96]: can be `revertsWithCustomError`

File: test/0.8.9/oracle/validators-exit-bus-oracle-access-control.test.js (hunk 1 above)
Comment (Jeday) [reply test/0.8.9/oracle/validators-exit-bus-oracle-access-control.test.js:87]: This seems very handy. Can you point to docs/package this is from? I got ambiguous results in google.

Comment (TheDZhon) [reply test/0.8.9/oracle/validators-exit-bus-oracle-access-control.test.js:87]: The motivational example https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/test/0.8.9/staking-router-deposits.test.js#L31,  and the actual `EvmSnapshot` implementation: https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/test/helpers/blockchain.js

`evm_snapshot` and `evm_revert` low-level methods both are from [ganache](https://docs.nethereum.com/en/latest/ethereum-and-clients/ganache-cli/), it's a common practice to use them for test fixtures, especially in brownie and ape.

**Review by TheDZhon — APPROVED** [review]
LGTM to merge into #624

**Pull request merged**